lib/asciidoctor/parser.rb

```diff
@@ -1234,7 +1235,10 @@ def self.catalog_inline_anchors text, document
     text.scan(InlineAnchorRx) do
       # lead with assignments for Ruby 1.8.7 compat
       id, reftext = $1 || $3, $2 || $4
-      document.register :ids, [id, reftext] unless $&.start_with? '\\'
+      unless $&.start_with? '\\'
+        document.register(:ids, [id, reftext])
+        document.register(:refs, [id, reftext])
```

Comment:
Unfortunately, until #61 is resolved, we don't have a node to store at this phase. We could consider using special type of Inline node that could hold the reftext and the parent node. I'll see what I can do.